spec/lib/serialization_core_spec.rb

```diff
@@ -1,82 +1,212 @@
 require 'spec_helper'
 
-describe FastJsonapi::ObjectSerializer do
-  include_context "movie class"
+describe FastJsonapi::SerializationCore do
+  include_context 'movie class'
   include_context 'group class'
 
-  context 'when testing class methods of serialization core' do
-    it 'returns correct hash when id_hash is called' do
-      inputs = [{id: 23, record_type: :movie}, {id: 'x', record_type: 'person'}]
-      inputs.each do |hash|
-        result_hash = MovieSerializer.send(:id_hash, hash[:id], hash[:record_type])
-        expect(result_hash[:id]).to eq hash[:id].to_s
-        expect(result_hash[:type]).to eq hash[:record_type]
+  describe '#self.id_hash' do
+    subject(:id_hash) { MovieSerializer.id_hash(id, 'movie') }
+
+    context 'when id is string' do
+      let(:id) { 'x' }
+
+      it 'returns id_hash' do
+        expect(id_hash[:id]).to eq 'x'
+        expect(id_hash[:type]).to eq 'movie'
+      end
+    end
+
+    context 'when id is int' do
+      let(:id) { 23 }
+
+      it 'turns id into string' do
+        expect(id_hash[:id]).to eq '23'
+        expect(id_hash[:type]).to eq 'movie'
       end
+    end
+
+    context 'when id is blank' do
+      let(:id) { nil }
 
-      result_hash = MovieSerializer.send(:id_hash, nil, 'movie')
-      expect(result_hash).to be nil
+      it 'returns nil' do
```

Comment:
No need for `it 'returns nil'`,  `is_expected.to be_nil` will give the same message. Can use `it { is_expected.to be_nil }`